content/guides/getstarted/cosmwasm-to-solana.md

```diff
@@ -0,0 +1,427 @@
+- Messages: Uses `InstantiateMsg`, `ExecuteMsg`, and `QueryMsg` to define
+  contract interactions.
+
+**Solana**
+
+- Execution Environment: Runs on the Solana blockchain, optimized for high
+  throughput and low latency.
+- State Management: Primarily uses `Borsh` for serialization and deserialization
+  of state stored in accounts. Solana recommends Borsh for its efficiency, but
+  developers can use other serialization formats like bincode if desired.
+- Entry Point: A single entry point (process_instruction) handles all
+  instructions.
+- Instructions: Uses custom-defined instructions encoded typically as byte
+  arrays, which are parsed within process_instruction to determine the specific
+  operations to perform. Instructions must be explicitly defined and handled
+  within the smart contract.
+
+## State Management
+
+In CosmWasm, state is stored in the smart contract's storage, which is managed
+by the Cosmos SDK. This contract interacts with this storage via helper
+functions provided by `cosmwasm-storage`. This storage is part of the blockchain
+state and is specific to each contract instance. State variables are typically
+defined in a `state.rs` file and are serialized/deserialized using `serde`.
+State management functions are used within the contract logic to read from and
+write to the storage.
+
+In contrast, Solana programs manage state by interacting with the data of
+specific accounts that are passed into the program during execution. These
+accounts can be created and assigned by the System Program and are allocated a
+certain amount of space for storing data. The state within these accounts is
+serialized and deserialized using libraries like Borsh. Solana programs are
+responsible for interpreting the account data as state information, writing to
+and reading from these accounts directly in the program logic.
+
+The core concept of accounts on Solana is crucial to understanding state
+management and effectively designing Solana programs. Each account on Solana can
+be associated with a specific program (smart contract), and only the owning
+program can modify its state. This model differs significantly from many other
+blockchain platforms where the state is managed more directly through the smart
+contract itself. For detailed information on the Solana Account Model, refer to
+the Solana documentation [here](https://solana.com/docs/core/accounts).
+
+### Key Differences
+
+1. Storage Abstraction:
+   - **CosmWasm**: Uses abstractions like `cosmwasm_storage` to manage state.
+     The contract interacts with a key-value storage system provided by the
+     Cosmos SDK.
+   - **Solana**: Directly reads from and writes to account data, which is part
+     of the account's state on the Solana blockchain.
+2. Serialization:
+   - **CosmWasm**: Typically uses `serde` for serialization of state structures.
+   - **Solana**: Uses `Borsh` for serialization and deserialization of account
```

Comment:
This isn't necessarily true. On Solana, serialization framework is a choice. You can literally roll your own if you want to. Anchor uses Borsh, but within Native you can use whatever you choose.

Reply:
oh cool! Updated 👍